src/main/java/com/minecraftabnormals/abnormals_core/core/api/conditions/loot/ConfigLootCondition.java

```diff
@@ -0,0 +1,114 @@
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * A loot condition that is registered and functions exactly the same way as {@link ConfigCondition}, but for loot tables instead.
+ *
+ * @author abigailfails
+ */
+public class ConfigLootCondition implements ILootCondition {
+    private final ForgeConfigSpec.ConfigValue<?> value;
+    private final String valueID;
+    private final Map<IConfigPredicate, Boolean> predicates;
+    private final boolean inverted;
+    private final ResourceLocation location;
+
+    public ConfigLootCondition(ResourceLocation location, ForgeConfigSpec.ConfigValue<?> value, String valueID, Map<IConfigPredicate, Boolean> predicates, boolean inverted) {
+        this.location = location;
+        this.value = value;
+        this.valueID = valueID;
+        this.predicates = predicates;
+        this.inverted = inverted;
+    }
+
+    @Override
+    public LootConditionType func_230419_b_() {
+        return Registry.LOOT_CONDITION_TYPE.getOrDefault(this.location);
+    }
+
+    @Override
+    public boolean test(LootContext context) {
+        boolean returnValue;
+        if (predicates.size() > 0) {
+            returnValue = this.predicates.keySet().stream().allMatch(c -> this.predicates.get(c) != c.test(value));
+        } else if (value.get() instanceof Boolean) {
+            returnValue = (Boolean) value.get();
+        } else throw new IllegalStateException("Predicates required for non-boolean ConfigLootCondition, but none found");
+        return this.inverted != returnValue;
+    }
+
+    public static class Serializer implements ILootSerializer<ConfigLootCondition> {
+        private final Map<String, ForgeConfigSpec.ConfigValue<?>> configValues;
+        private final ResourceLocation location;
+
+        public Serializer(String modId, Map<String, ForgeConfigSpec.ConfigValue<?>> configValues) {
+            this.location = new ResourceLocation(modId, "config");
+            this.configValues = configValues;
+        }
+
+        @Override
+        public void serialize(JsonObject json, ConfigLootCondition value, JsonSerializationContext context) {
+            json.addProperty("value", value.valueID);
+            if (!value.predicates.isEmpty()) {
+                JsonArray predicates = new JsonArray();
+                json.add("predicates", predicates);
+                for (Map.Entry<IConfigPredicate, Boolean> predicatePair : value.predicates.entrySet()) {
+                    IConfigPredicate predicate = predicatePair.getKey();
+                    ResourceLocation predicateID = predicate.getID();
+                    JsonObject object = new JsonObject();
+                    predicates.add(object);
+                    object.addProperty("type", predicateID.toString());
+                    ConfigCondition.Serializer.CONFIG_PREDICATE_SERIALIZERS.get(predicateID).write(object, predicate);
+                    object.addProperty("inverted", predicatePair.getValue());
+                }
```

Comment:
Small style suggestions. The formatting in this suggestion may be a bit messed up.
```suggestion
                JsonArray predicates = new JsonArray();
                for (Map.Entry<IConfigPredicate, Boolean> predicatePair : value.predicates.entrySet()) {
                    IConfigPredicate predicate = predicatePair.getKey();
                    ResourceLocation predicateID = predicate.getID();
                    JsonObject object = new JsonObject();
                    object.addProperty("type", predicateID.toString());
                    ConfigCondition.Serializer.CONFIG_PREDICATE_SERIALIZERS.get(predicateID).write(object, predicate);
                    object.addProperty("inverted", predicatePair.getValue());
                    predicates.add(object);
                }
                json.add("predicates", predicates);
```